From 646f6ef30e4b01b1f2ebbdb8ab507ee7d174a460 Mon Sep 17 00:00:00 2001
From: Didier Roche <didrocks@ubuntu.com>
Date: Thu, 7 Oct 2010 16:39:33 +0200
Subject: [PATCH] add a function to expose GtkMenuShell::active_menu_item

We can have some cases where we want to know which is the selected
menuitem in a GtkMenuShell before it became activated.
Adding a gtk_menu_shell_get_selected_item to the API to get it.
---
 gtk/gtk.symbols    |    1 +
 gtk/gtkmenushell.c |   21 +++++++++++++++++++++
 gtk/gtkmenushell.h |    1 +
 3 files changed, 23 insertions(+), 0 deletions(-)

Index: gtk+-2.91.6/gtk/gtk.symbols
===================================================================
--- gtk+-2.91.6.orig/gtk/gtk.symbols	2010-12-07 16:33:55.087103003 +1100
+++ gtk+-2.91.6/gtk/gtk.symbols	2010-12-07 16:34:02.207103003 +1100
@@ -1417,6 +1417,7 @@
 gtk_menu_shell_cancel
 gtk_menu_shell_deactivate
 gtk_menu_shell_deselect
+gtk_menu_shell_get_selected_item
 gtk_menu_shell_get_take_focus
 gtk_menu_shell_get_type G_GNUC_CONST
 gtk_menu_shell_insert
Index: gtk+-2.91.6/gtk/gtkmenushell.c
===================================================================
--- gtk+-2.91.6.orig/gtk/gtkmenushell.c	2010-12-07 16:33:59.487103001 +1100
+++ gtk+-2.91.6/gtk/gtkmenushell.c	2010-12-07 16:34:02.207103003 +1100
@@ -1348,6 +1348,27 @@
     }
 }
 
+/**
+ * gtk_menu_shell_get_selected_item:
+ * @menu_shell: a #GtkMenuShell
+ *
+ * Retrieves a pointer to the selected menuitem, or %NULL
+ * if there is no menuitem selected. A selected menuitem is
+ * one that is displayed in the prelight state for that menu.
+ *
+ * Return value: a #GtkWidget or %NULL in case there is no
+ * selected menuitem
+ *
+ * Since: 2.24
+ **/
+GtkWidget *
+gtk_menu_shell_get_selected_item (GtkMenuShell *menu_shell)
+{
+  g_return_val_if_fail (GTK_IS_MENU_SHELL (menu_shell), NULL);
+
+  return menu_shell->active_menu_item;
+}
+
 void
 gtk_menu_shell_activate_item (GtkMenuShell      *menu_shell,
 			      GtkWidget         *menu_item,
Index: gtk+-2.91.6/gtk/gtkmenushell.h
===================================================================
--- gtk+-2.91.6.orig/gtk/gtkmenushell.h	2010-12-06 02:46:52.000000000 +1100
+++ gtk+-2.91.6/gtk/gtkmenushell.h	2010-12-07 16:34:02.207103003 +1100
@@ -114,6 +114,7 @@
 void  gtk_menu_shell_activate_item     (GtkMenuShell *menu_shell,
 					GtkWidget    *menu_item,
 					gboolean      force_deactivate);
+GtkWidget *gtk_menu_shell_get_selected_item (GtkMenuShell *menu_shell);
 void  gtk_menu_shell_select_first      (GtkMenuShell *menu_shell,
 					gboolean      search_sensitive);
 void _gtk_menu_shell_select_last       (GtkMenuShell *menu_shell,
